Valet/VALValet.m

```diff
@@ -227,15 +227,25 @@ - (BOOL)isEqualToValet:(VALValet *)otherValet;
 
 - (BOOL)canAccessKeychain;
 {
-    NSString *canaryKey = @"VAL_KeychainCanaryUsername";
-    NSString *canaryValue = @"VAL_KeychainCanaryPassword";
+    BOOL canAccessKeychain = NO;
     
-    NSMutableDictionary *query = [self.baseQuery mutableCopy];
-    [query addEntriesFromDictionary:[self _secItemFormatDictionaryWithKey:canaryKey]];
-    [query addEntriesFromDictionary:@{ (__bridge id)kSecValueData : [canaryValue dataUsingEncoding:NSUTF8StringEncoding] }];
+    [self.lockForSetAndRemoveOperations lock];
```

Comment:
Probably should be in a `@try`/`@finally` to make sure its unlocked if an exception is thrown.